be/src/exec/pipeline/pipeline_builder.cpp

```diff
@@ -103,23 +108,25 @@ OpFactories PipelineBuilderContext::maybe_interpolate_local_shuffle_exchange(
     return operators_source_with_local_shuffle;
 }
 
-OpFactories PipelineBuilderContext::maybe_gather_pipelines_to_one(std::vector<OpFactories>& pred_operators_list) {
+OpFactories PipelineBuilderContext::maybe_gather_pipelines_to_one(RuntimeState* state,
+                                                                  std::vector<OpFactories>& pred_operators_list) {
     // If there is only one pred pipeline, we needn't local passthrough anymore.
     if (pred_operators_list.size() == 1) {
         return pred_operators_list[0];
     }
 
-    // Approximately, each pred driver can output config::vector_chunk_size rows at the same time.
+    // Approximately, each pred driver can output state->chunk_size() rows at the same time.
     size_t max_row_count = 0;
     for (const auto& pred_ops : pred_operators_list) {
         auto* source_operator = down_cast<SourceOperatorFactory*>(pred_ops[0].get());
-        max_row_count += source_operator->degree_of_parallelism() * config::vector_chunk_size;
+        max_row_count += source_operator->degree_of_parallelism() * state->chunk_size();
     }
 
     auto pseudo_plan_node_id = next_pseudo_plan_node_id();
     auto mem_mgr = std::make_shared<LocalExchangeMemoryManager>(max_row_count);
     auto local_exchange_source =
             std::make_shared<LocalExchangeSourceOperatorFactory>(next_operator_id(), pseudo_plan_node_id, mem_mgr);
+    local_exchange_source->set_runtime_state(state);
```

Comment:
why not set in constructor ?

Reply:
Have been set in OperatorFactory::prepare, I will remove this.